### PR TITLE
[videotoolbox] Add nullability to (generated and manual) bindings

### PR DESCRIPTION
**Pull request by tj-devel709**
This PR aims to bring nullability changes to VideoToolbox.
Following the steps [here](https://github.com/xamarin/xamarin-macios/wiki/Nullability):

1. I am adding `nullable enable` to all manual files that are not "API_SOURCES" in src/frameworks.sources and making the required nullability changes
2. Changing all `throw new ArgumentNullException ("object"));` to `ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (object));` for size saving optimization as well to mark that this framework contains nullability changes
3. Changing any `== null` or `!= null` to `is null` and `is not null`

### CODE REVIEW

File: src/VideoToolbox/VTCompressionProperties.cs [hunk 1]
```diff
@@ -508,7 +511,7 @@ public partial class VTCompressionProperties {
 		[Introduced (PlatformName.TvOS, 13,0)]
 #endif
 		public VTAlphaChannelMode AlphaChannelMode {
-			get => VTAlphaChannelModeExtensions.GetValue (GetNSStringValue (VTCompressionPropertyKey.AlphaChannelMode));
+			get => VTAlphaChannelModeExtensions.GetValue (GetNSStringValue (VTCompressionPropertyKey.AlphaChannelMode)!);
```
Comment (mandel-macaque) [inline src/VideoToolbox/VTCompressionProperties.cs:514]: Why not setting the property as nullable?

File: src/VideoToolbox/VTVideoEncoder.cs [hunk 2]
```diff
@@ -166,52 +173,54 @@ internal VTVideoEncoder (NSDictionary dict)
...
+			if (constant is not null) {
 				if (dict.TryGetValue (constant, out NSDictionary d)) // optional
 					SupportedSelectionProperties = d;
 			}
 
 			constant = VTVideoEncoderList.PerformanceRating;
-			if (constant != null) {
-				PerformanceRating = dict [constant] as NSNumber; // optional
+			if (constant is not null) {
+				if (dict [constant] is NSNumber performanceRatingNum)
```
Comment (mandel-macaque) [inline src/VideoToolbox/VTVideoEncoder.cs:189]: Merge them in and &&

File: src/VideoToolbox/VTVideoEncoder.cs [hunk 3]
```diff
@@ -166,52 +173,54 @@ internal VTVideoEncoder (NSDictionary dict)
...
+					PerformanceRating = performanceRatingNum; // optional
 			}
 
 			constant = VTVideoEncoderList.QualityRating;
-			if (constant != null) {
-				QualityRating = dict [constant] as NSNumber; // optional
+			if (constant is not null) {
+				if (dict [constant] is NSNumber qualityRatingNum)
```
Comment (mandel-macaque) [inline src/VideoToolbox/VTVideoEncoder.cs:195]: Same.

**Review by mandel-macaque — CHANGES_REQUESTED** [review]

File: src/VideoToolbox/VTVideoEncoder.cs (hunk 2 above)
Comment (rolfbjarne) [reply src/VideoToolbox/VTVideoEncoder.cs:189]: `PerformanceRating` is nullable (`NSNumber?`), so why is this change needed?

**Comment by tj-devel709** [comment]
test failures are unrelated timeouts!

File: src/VideoToolbox/VTCompressionProperties.cs [hunk 4]
```diff
@@ -507,9 +510,20 @@ public partial class VTCompressionProperties {
 		[Introduced (PlatformName.iOS, 13,0)]
 		[Introduced (PlatformName.TvOS, 13,0)]
 #endif
-		public VTAlphaChannelMode AlphaChannelMode {
-			get => VTAlphaChannelModeExtensions.GetValue (GetNSStringValue (VTCompressionPropertyKey.AlphaChannelMode));
-			set => SetStringValue (VTCompressionPropertyKey.AlphaChannelMode, value.GetConstant ());
+		public VTAlphaChannelMode? AlphaChannelMode {
```
Comment (rolfbjarne) [inline src/VideoToolbox/VTCompressionProperties.cs:513]: This is a breaking change, you can't make a value type (enum or struct) nullable :/

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: src/VideoToolbox/VTCompressionProperties.cs (hunk 4 above)
Comment (tj-devel709) [reply src/VideoToolbox/VTCompressionProperties.cs:513]: Okay! So I will revert the changes to this property

**Review by rolfbjarne — APPROVED** [review]

File: src/VideoToolbox/VTCompressionProperties.cs (hunk 1 above)
Comment (tj-devel709) [reply src/VideoToolbox/VTCompressionProperties.cs:514]: I applied this change but reverted it due to Rolf's comment below

**Pull request merged**